pkg/agent/plugin/nodeattestor/k8s/psat/psat.go

```diff
@@ -83,8 +87,17 @@ func (p *AttestorPlugin) FetchAttestationData(stream nodeattestor.NodeAttestor_F
 		return psatError.New("fail to get claims from token: %v", err)
 	}
 
-	if claims.K8s.Pod.UID == "" {
-		return psatError.New("token claim pod UID is empty")
+	if claims.K8s.Namespace == "" {
+		return psatError.New("token claim namespace is empty")
+	}
+
+	if claims.K8s.Pod.Name == "" {
+		return psatError.New("token claim pod name is empty")
+	}
+
+	nodeName, err := p.k8sClient.GetNode(claims.K8s.Namespace, claims.K8s.Pod.Name)
```

Comment:
hmm. wonder if we should be getting this from the kubelet instead of requiring agents to have direct access to the API server? either that or we do the work to remove the requirement for agent's to provide the SPIFFE ID. @evan2645 , thoughts?

Reply:
I have been thinking about this and asked @evan2645  for some thoughts. It sounds good to me to get this info from kubelet :+1: